payjoin/src/receive/multiparty/mod.rs

```diff
@@ -253,51 +301,124 @@ impl FinalizedProposal {
 mod test {
 
     use std::any::{Any, TypeId};
+    use std::str::FromStr;
 
-    use payjoin_test_utils::{BoxError, PARSED_ORIGINAL_PSBT};
+    use bitcoin::Psbt;
+    use payjoin_test_utils::{BoxError, MULTIPARTY_SENDER_ONE, MULTIPARTY_SENDER_TWO};
 
-    use super::{v1, v2, FinalizedProposal, UncheckedProposalBuilder, SUPPORTED_VERSIONS};
+    use super::error::IdenticalProposalError;
+    use super::{
+        v1, v2, FinalizedProposal, InternalMultipartyError, MultipartyError,
+        UncheckedProposalBuilder, SUPPORTED_VERSIONS,
+    };
     use crate::receive::optional_parameters::Params;
-    use crate::receive::v2::test::SHARED_CONTEXT;
+    use crate::receive::v2::test::{MULTIPARTY_CONTEXT, SHARED_CONTEXT};
 
-    fn multiparty_proposal_from_test_vector() -> v1::UncheckedProposal {
+    fn multiparty_proposal_one() -> v1::UncheckedProposal {
         let pairs = url::form_urlencoded::parse("v=2&optimisticmerge=true".as_bytes());
         let params = Params::from_query_pairs(pairs, SUPPORTED_VERSIONS)
             .expect("Could not parse from query pairs");
-        v1::UncheckedProposal { psbt: PARSED_ORIGINAL_PSBT.clone(), params }
+        v1::UncheckedProposal {
+            psbt: Psbt::from_str(MULTIPARTY_SENDER_ONE).expect("known psbt should parse"),
+            params,
+        }
+    }
+
+    fn multiparty_proposal_two() -> v1::UncheckedProposal {
```

Comment:
More of a question: is it worth it to just consolidate this method with the one above and return a Vec of test vectors? I only say this because they seem to share some boilerplate code.
```rust
fn multiparty_proposals() -> Vec<v1::UncheckedProposal> {
    let query = "v=2&optimisticmerge=true";
    let pairs = url::form_urlencoded::parse(query.as_bytes());
    let params = Params::from_query_pairs(pairs, SUPPORTED_VERSIONS)
        .expect("Could not parse from query pairs");

    [MULTIPARTY_SENDER_ONE, MULTIPARTY_SENDER_TWO]
        .iter()
        .map(|psbt_str| v1::UncheckedProposal {
            psbt: Psbt::from_str(psbt_str).expect("known psbt should parse"),
            params,
        })
        .collect()
}
```